addons/mail/static/src/models/autocomplete_input_view/autocomplete_input_view.js

```diff
@@ -0,0 +1,232 @@
+        },
+        onComponentUpdate() {
+            if (this.doFocus) {
+                this.component.root.el.focus();
+                this.update({ doFocus: false });
+            }
+        },
+        onFocusin() {
+            if (!this.exists()) {
+                return;
+            }
+            this.update({ isFocused: true });
+        },
+        /**
+         * @param {MouseEvent} ev
+         */
+        onKeydown(ev) {
+            if (this.chatWindowOwnerAsNewMessageFormInputOwner) {
+                return;
+            }
+            if (ev.key === 'Escape') {
+                this.delete();
+            }
+        },
+        /**
+         * Called when selecting an item in this autocomplete input view
+         *
+         * @param {Event} ev
+         * @param {Object} ui
+         * @param {Object} ui.item
+         * @param {integer} ui.item.id
+         */
+        async onSelect(ev, ui) {
+            if (this.chatWindowOwnerAsNewMessageFormInputOwner) {
+                const chat = await this.messaging.getChat({ partnerId: ui.item.id });
+                if (!chat) {
+                    return;
+                }
+                this.messaging.chatWindowManager.openThread(chat, {
+                    makeActive: true,
+                    replaceNewMessage: true,
+                });
+            }
+            if (this.discussAsChannelMobileAddItemHeaderOwner) {
+                this.discussAsChannelMobileAddItemHeaderOwner.handleAddChannelAutocompleteSelect(ev, ui);
+            }
+            if (this.discussAsChatMobileAddItemHeaderOwner) {
+                this.discussAsChatMobileAddItemHeaderOwner.handleAddChatAutocompleteSelect(ev, ui);
+            }
+            if (this.discussSidebarCategoryAsAddingItemInputViewOwner) {
+                return this.discussSidebarCategoryAsAddingItemInputViewOwner.onAddItemAutocompleteSelect(ev, ui);
+            }
+            if (this.messagingMenuAsMobileNewMessageInput) {
+                this.messaging.openChat({ partnerId: ui.item.id });
+            }
+        },
+        /**
+         * Called when typing in this autocomplete input view.
+         *
+         * @param {Object} req
+         * @param {string} req.term
+         * @param {function} res
+         */
+        onSource(req, res) {
+            if (this.chatWindowOwnerAsNewMessageFormInputOwner) {
+                this.messaging.models['Partner'].imSearch({
+                    callback: (partners) => {
+                        const suggestions = partners.map(partner => {
+                            return {
+                                id: partner.id,
+                                value: partner.nameOrDisplayName,
+                                label: partner.nameOrDisplayName,
+                            };
+                        });
+                        res(_.sortBy(suggestions, 'label'));
+                    },
+                    keyword: _.escape(req.term),
```

Comment:
To double check for security, but I'm not sure to understand why we escape here, if the only case with is html is channel from discuss sidebar. It looks like this would double escape with the t-esc and therefor wrongly show escape chars.